src-front/src/lib/vectorsUtility.ts

```diff
@@ -0,0 +1,70 @@
+export class Vector {
+  constructor(
+    public x: number,
+    public y: number,
+  ) {
+    this.x = x;
+    this.y = y;
+  }
+
+  add(other: Vector): Vector {
+    return new Vector(this.x + other.x, this.y + other.y);
+  }
+
+  mult(coef: number): Vector {
+    return new Vector(this.x * coef, this.y * coef);
+  }
+}
+
+export interface Point {
+  x: number;
+  y: number;
+}
+
+export function calculateVec(
+  x1: number,
+  y1: number,
+  x2: number,
+  y2: number,
+  coef: number,
+) {
+  const x = (x2 - x1) * coef;
+  const y = (y2 - y1) * coef;
+
+  const res = new Vector(x, y);
+
+  return res;
+}
+
+export function calculateVecSquared(
```

Comment:
If the performance hit is heavy, I would seriously consider memoization with `Map` (more performant than a simple object used as a dictionary). Looking at the code, `Math.hypot()` seem to also be a potential solution for an additional speed boost according to this StackOverflow post: https://stackoverflow.com/questions/37425060/javascript-math-sqrt-performance

In all cases, benchmarks and the browser profiler would be needed to figure out what is the best approach, and make sure the bottleneck isn't somewhere else.